cluster-lifecycle/update-managedcluster-status-to-include-clusterinfo.md

```diff
@@ -0,0 +1,254 @@
+
+## Motivation
+
+Today, several label properties, including clusterID, cloud and vendor, are updated on ManagedCluster objects that can be modified by users, creating invalid data for matching behavior. It would be a good idea to include cluster info provided by the Klusterlet in the status of ManagedCluster.
+
+It is required from other components that more cluster specific info should be added into ManagedCluster API. This information includes the clusterID, the ocp distribution version, vendor and cloud provider etc. On another hand, ManagedCluster API should be designed to have a generic and extensible structure to include this info that also can work with different implementations.
+
+### Goals
+
+- Allow extensible info in managed cluster to be added in ManagedCluster API
+- Remove cluster info lables from managed cluster eventually and use managed cluster status instead;
+- Align wth upstream design;
+
+### Non-Goals
+
+- Replace and deprecate ManagedClusterInfo API
+
+## Proposal
+
+To align with the upstream proposal for ClusterID (https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit), leverage ClusterClaims created on managed cluster to store and manage cluster information.
+
+### Risks and Mitigations
+
+- The upstream proposal for ClusterID is not finalized yet and may change in the future.  
+The mitigation is to create ClusterClaim CRD in group cluster.open-cluster-management.io with version v1alph1 for now. Once the proposal is finalized, support both of them for one or two releases and then deprecate the one in group cluster.open-cluster-management.io.
+
+## Design Details
+
+As shown in the table below, ManagedClusterInfo includes 10 fields in its status. Some of them will NOT be merged into ManagedCluster.
+
+| Name    | Description    | Comments   |
+| ---- | ---- | ---- |
+| Conditions | condition information for a managed cluster | **Not merged**. ManagedCluster has its own Conditions. |
+| Version | the kube version of managed cluster | **Not merged**. ManagedCluster has a field with same name in its status, which also points to the kube version of managed cluster |
+| KubeVendor | the kubernetes provider of the managed cluster| |
+| CloudVendor | the cloud provider for the managed cluster | |
+| ClusterID | the identifier of managed cluster | |
+| DistributionInfo | the information about distribution of managed cluster | **Not merged**. It's product specific. |
+| ConsoleURL | the url of console in managed cluster | ？ |
+| NodeList | a list of the status of nodes | **Not merged**. There is a scale issue for a cluster with a large number of nodes. |
+| LoggingEndpoint | the endpoint to connect to logging server of managed cluster | **Not merged**. It's product specific. |
+| LoggingPort | the port to connect to logging server of managed cluster | **Not merged**. It's product specific. |
```

Comment:
Do we need to carry any information here about what type of logging is being used?

Reply:
logging is something we used in ACM internally, this is not supposed to be used externally so we may not expose it as a clusterclaim.

Reply:
Currently, ManagedClusterInfo includes LoggingEndpoint and LoggingPort in its status, which are ACM specific fields to access logs of pods on managed cluster . Their values are independent of Kubernetes distributions and vendors. So we are not going to create ClusterClaims for them.